src/ctl.c

```diff
@@ -1971,6 +1979,41 @@ thread_prof_active_ctl(tsd_t *tsd, const size_t *mib,
 	return ret;
 }
 
+static int
+thread_idle_ctl(tsd_t *tsd, const size_t *mib,
+    size_t miblen, void *oldp, size_t *oldlenp, void *newp,
+    size_t newlen) {
+	int ret;
+
+	/* See the comment in thread_tcache_flush_ctl. */
+	READONLY();
+	WRITEONLY();
+
+	if (tcache_available(tsd)) {
+		tcache_flush(tsd);
+	}
+	/*
+	 * This heuristic is perhaps not the most well-considered.  But it
+	 * matches the only idling policy we have experience with in the status
+	 * quo.  Over time we should investigate more principled approaches.
+	 */
+	if (opt_narenas > ncpus * 2) {
```

Comment:
I'd say use the thread # approach instead? I think that heuristic was more of a way saying "if the malloc_conf is tuned".